src/main/java/org/togetherjava/formatter/tokenizer/Lexer.java

```diff
@@ -0,0 +1,48 @@
+package org.togetherjava.formatter.tokenizer;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+
+public class Lexer {
+    private static final Lexer instance = new Lexer();
+
+    public static Lexer getInstance() {
+        return instance;
+    }
+
+    public List<Token> tokenize(String input) {
+        return tokenize(Arrays.asList(input.split("\n")));
+    }
+
+    public List<Token> tokenize(List<String> lines) {
```

Comment:
Break this up into multiple functions, the cyclomatic complexity is pretty high here. Put that lambda in its own method and reference it. Maybe break even that more up into methods to keep the complexity low.